### PR TITLE
hardening/793_move_attrpersistence_to_orion_mongo_sink

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue https://github.com/telefonicaid/fiware-cygnus/issues/793
* 100% unit test passed:
```
Results : Tests run: 88, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e test passed: 
`OrionMongoSink` (Checking with `attr_persistence` = filas. Result must be `column` persistence with the actual check):
```
time=2016-02-16T14:09:29.381CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoSink[78] : [mongo-sink] Reading configuration (attr_persistence=column)
```
`OrionSTHSink` (Without `attr_persistence`, message doesn't appear):
```
time=2016-02-16T14:12:54.476CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[113] : [sth-sink] Reading configuration (mongo_hosts=localhost:27017)
time=2016-02-16T14:12:54.476CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[115] : [sth-sink] Reading configuration (mongo_username=)
time=2016-02-16T14:12:54.476CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[118] : [sth-sink] Reading configuration (mongo_password=)
time=2016-02-16T14:12:54.476CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[120] : [sth-sink] Reading configuration (db_prefix=sth_)
time=2016-02-16T14:12:54.477CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[122] : [sth-sink] Reading configuration (collection_prefix=sth_)
time=2016-02-16T14:12:54.482CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionMongoBaseSink[124] : [sth-sink] Reading configuration (should_hash=false)
time=2016-02-16T14:12:54.483CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[163] : [sth-sink] Invalid configuration (data_model=dm-by-afrentity)
time=2016-02-16T14:12:54.484CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[168] : [sth-sink] Reading configuration (enable_grouping=true)
time=2016-02-16T14:12:54.484CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[171] : [sth-sink] Reading configuration (batch_size=0)
time=2016-02-16T14:12:54.484CET | lvl=DEBUG | trans= | srv= | subsrv= | function=configure | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[174] : [sth-sink] Reading configuration (batch_timeout=0)
```

* Assignee @frbattid

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -17,3 +17,4 @@
 - [HARDENING] Remove the persistOne method from all the sinks (#609)
 - [HARDENING] Invalidate the sink if the data_model parameter is wrong (and prepare the code for further parameter checks) (#718)
 - [BUG] Fix the generalized Hive-like encoding style used in OrionHDFSSink (#781)
+- [HARDENING] Relocation of attr_persistence check from OrionMongoBaseSink to OrionMongoSink (#793)
```
Comment (frbattid) [inline CHANGES_NEXT_RELEASE]: Relocate attr_persistence...

File: src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoBaseSink.java [hunk 2]
```diff
@@ -122,9 +122,6 @@ public void configure(Context context) {
         LOGGER.debug("[" + this.getName() + "] Reading configuration (collection_prefix=" + collectionPrefix + ")");
         shouldHash = context.getBoolean("should_hash", false);
         LOGGER.debug("[" + this.getName() + "] Reading configuration (should_hash=" + shouldHash + ")");
-        this.rowAttrPersistence = context.getString("attr_persistence", "row").equals("row");
```
Comment (frbattid) [inline src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoBaseSink.java]: Now, `rowAttrPersistence` must be declared at `OrionMongoSink`, not at `OrionMongoBaseSink`

File: src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java [hunk 3]
```diff
@@ -69,6 +72,19 @@ void persistBatch(Batch batch) throws Exception {
...
     
+    @Override
+    public void configure (Context context) {
+        this.rowAttrPersistence = context.getString("attr_persistence", "row").equals("row");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (attr_persistence="
+                + (this.rowAttrPersistence ? "row" : "column") + ")");
+        super.configure(context);
+    } // configure
+    
+    @Override
```
Comment (frbattid) [inline src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java]: I think this is unnecessary.

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (pcoello25) [inline CHANGES_NEXT_RELEASE]: Fixed in 1dbbacd

File: src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoBaseSink.java (hunk 2 above)
Comment (pcoello25) [inline src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoBaseSink.java]: Fixed in 1dbbacd

File: src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java (hunk 3 above)
Comment (pcoello25) [inline src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java]: Fixed in 5aaf3de

**Comment by frbattid** [comment]
LGTM

**Pull request merged**